File: implementation/Scala/src/main/java/synapses/jvm/library/Statistics.java
package synapses.jvm.library;

import synapses.JavaInterface.Statistics$;

import java.util.stream.Stream;

public class Statistics {

    public static double rootMeanSquareError(Stream<double[][]> expectedValuesWithOutputValues){
        return Statistics$.MODULE$.rootMeanSquareError(expectedValuesWithOutputValues);
    }

}

File: implementation/CSharp/Statistics.cs
using System;
using System.Collections.Generic;
using System.Linq;
using FSharpx.Collections;
using Microsoft.FSharp.Collections;

namespace SynapsesCSharp
{
    public class Statistics
    {
        
        public static double rootMeanSquareError(IEnumerable<(double [], double [])> expectedValuesWithOutputValues)
        {
            LazyList<Tuple<FSharpList<double>, FSharpList<double>>> expWithOuts = LazyList
                .ofSeq(expectedValuesWithOutputValues.Select(exOut =>
                    new Tuple<FSharpList<double>, FSharpList<double>>(
                        ListModule.OfSeq(exOut.Item1),
                        ListModule.OfSeq(exOut.Item2)
                        )
                    )
                );
            return Synapses.Statistics.rootMeanSquareError(expWithOuts);
        }
        
    }
}